src/screens/SUE/SueMapViewScreen.tsx

```diff
@@ -36,17 +42,19 @@ export const SueMapViewScreen = ({
   const [selectedPosition, setSelectedPosition] = useState<
     Location.LocationObjectCoords | undefined
   >(position);
-  const [locationsWithPin, setLocationsWithPin] = useState(locations);
   const [updatedRegion, setUpdatedRegion] = useState<boolean>();
 
   const iconName = 'location';
 
-  useEffect(() => {
-    setLocationsWithPin((prevData) =>
-      prevData.map((item) =>
-        item?.iconName === iconName ? { iconName, position: selectedPosition } : item
-      )
-    );
+  const locationsWithPin = useMemo(() => {
+    if (selectedPosition) {
+      return [
+        ...locations.filter((item) => item.id !== SELECTED_MARKER_ID), // Remove old pin
+        { iconName, position: selectedPosition, id: SELECTED_MARKER_ID } // Add new pin
```

Comment:
with seeing this again, why are we filtering all but the one and adding the one again? couldn't we update the position of the selected marker inside the locations?

Reply:
Directly mutating the position like `locations.find(...).position = selectedPosition` won’t trigger a re-render because React won’t detect the change if the reference to the locations array stays the same. Using `useMemo` with a new array ensures immutability and proper reactivity, so the component updates correctly.

Reply:
ok, too bad, but I thought so